### PR TITLE
chore(external docs): Add `kubernetes_logs` source

### PR DESCRIPTION
**Pull request by binarylogic**

### CODE REVIEW

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+			required:    false
+			type: string: {
+				default: "${VECTOR_SELF_NODE_NAME}"
+			}
+		}
+	}
+
+	output: logs: line: {
+		description: "fill in"
+		fields: {}
```
Comment (binarylogic) [inline docs/reference/components/sources/kubernetes_logs.cue:125]: @MOZGIII can you fill these out? The other sources have examples.

File: docs/reference/components/sources/kubernetes_logs.cue [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	configuration: {
+		annotation_fields: {
+			common:      false
+			description: "Configuration for how the events are annotated with Pod metadata."
+			required:    false
+			type: object: {
+				examples: []
+				options: {
```
Comment (binarylogic) [inline docs/reference/components/sources/kubernetes_logs.cue:47]: @MOZGIII can you make sure the options here are still accurate given the recent changes?

**Comment by binarylogic** [comment]
@MOZGIII I'm going to merge this to unblock some website work. Could you open a separate PR to finish off the docs? Thanks.

**Comment by MOZGIII** [comment]
> @MOZGIII I'm going to merge this to unblock some website work. Could you open a separate PR to finish off the docs? Thanks.

Sounds good!

**Pull request merged**